packages/mdc-dialog/foundation.ts

```diff
@@ -176,26 +176,34 @@ export class MDCDialogFoundation extends MDCFoundation<MDCDialogAdapter> {
     });
   }
 
-  handleInteraction(evt: MouseEvent | KeyboardEvent) {
-    const isClick = evt.type === 'click';
-    const isEnter = (evt as KeyboardEvent).key === 'Enter' || (evt as KeyboardEvent).keyCode === 13;
-    const isSpace = (evt as KeyboardEvent).key === 'Space' || (evt as KeyboardEvent).keyCode === 32;
+  handleClick(evt: MouseEvent) {
     const isScrim = this.adapter_.eventTargetMatches(evt.target, strings.SCRIM_SELECTOR);
-    const isDefault = !this.adapter_.eventTargetMatches(evt.target, strings.SUPPRESS_DEFAULT_PRESS_SELECTOR);
-
-    // Check for scrim click first since it doesn't require querying ancestors
-    if (isClick && isScrim && this.scrimClickAction_ !== '') {
+    // Check for scrim click first since it doesn't require querying ancestors.
+    if (isScrim && this.scrimClickAction_ !== '') {
       this.close(this.scrimClickAction_);
-    } else if (isClick || isSpace || isEnter) {
+    } else {
       const action = this.adapter_.getActionFromEvent(evt);
       if (action) {
         this.close(action);
-      } else if (isEnter && isDefault) {
-        this.adapter_.clickDefaultButton();
       }
     }
   }
 
+  handleKeydown(evt: KeyboardEvent) {
+    const isEnter = evt.key === 'Enter' || evt.keyCode === 13;
+    const isSpace = evt.key === 'Space' || evt.keyCode === 32;
+    if (!isSpace && !isEnter) {
+      return;
+    }
+
+    const isDefault = !this.adapter_.eventTargetMatches(
+        evt.target, strings.SUPPRESS_DEFAULT_PRESS_SELECTOR);
+    const action = this.adapter_.getActionFromEvent(evt);
```

Comment:
Let's add separate if condition for checking `action` event to reason that it has been handled in click callback.

```ts
// Action button callback is handled in `handleClick`.
if (action) {
  return;
}

if (isEnter && isDefault) {
  this.adapter_.clickDefaultButton();
}
```

Reply:
Great point, thanks. Done.